client/blocks/props-viewer/index.jsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * External Dependencies
+ */
+import React, { PropTypes, PureComponent } from 'react';
+
+/**
+ * Internal Dependencies
+ */
+import Card from 'components/card';
+import Gridicon from 'components/gridicon';
+
+// todo: move this file!
+const components = require( '../../../server/devdocs/proptypes-index.json' );
+
+/**
+ * Finds a non-example component in the library of components
+ * @param {string} slug The slug to search for
+ * @return {Array} An array of component matches
+ */
+export function findRealComponent( slug ) {
+	// remove the last character. As of right now, all plural display names are with just an 's'
+	const singular = slug.slice( 0, -1 );
+	return components.filter( ( component ) => {
+		return ( slug === component.slug || singular === component.slug ) && component.includePath.indexOf( 'example' ) < 0;
+	} );
+}
+
+/**
+ * Renders a table of prop-types for auto-documentation
+ */
+class PropsViewer extends PureComponent {
+	static displayName = 'PropsViewer';
+
+	static propTypes = {
+		/**
+		 * The slug of the component being displayed
+		 */
+		component: PropTypes.string.isRequired,
+
+		/**
+		 * The element to display as an example of this component
+		 */
+		example: PropTypes.element.isRequired
+	}
+
+	constructor( props ) {
+		super( props );
+	}
+
+	/**
+	 * Set the state of this component to the first matching slug
+	 * @param {string} slug The slug to search for
+	 */
+	setComponent = ( slug ) => {
+		let component = findRealComponent( slug );
+		if ( component.length > 0 ) {
+			component = component[ 0 ];
```

Comment:
Instead of applying `Array#filter` to the entire set of components only to pick the first matched, seems an appropriate use case for [`_.find`](https://lodash.com/docs/4.17.4#find) (there's a native [`Array#find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) but we can't use it since we don't polyfill ES2015+ prototype methods).